libs/wingsdk/src/target-tf-aws/bucket.ts

```diff
@@ -46,20 +47,54 @@ export const BUCKET_PREFIX_OPTS: NameOptions = {
   suffix: "-",
 };
 
+/**
+ * Properties for a new tfaws.Bucket.
+ */
+export interface BucketProps extends cloud.BucketProps {
+  /**
+   * This internal property is used to override the default bucket creation
+   * and instead use an existing bucket.
+   *
+   * @internal
+   */
+  _existingBucket?: DataAwsS3Bucket;
+}
+
 /**
  * AWS implementation of `cloud.Bucket`.
  *
  * @inflight `@winglang/sdk.cloud.IBucketClient`
  */
 export class Bucket extends cloud.Bucket implements IAwsBucket {
-  private readonly bucket: S3Bucket;
-  private readonly public: boolean;
+  /**
+   * Create a new cloud.Bucket from an existing s3 bucket.
+   *
+   * @param scope scope to use for the bucket data lookup
+   * @param bucketName the name of the existing bucket to reference
+   * @returns A cloud.Bucket
+   */
+  public static fromBucketName(scope: Construct, bucketName: string): Bucket {
+    const bucketLookup = new DataAwsS3Bucket(
+      scope,
+      `${bucketName}_${scope.node.addr.slice(-8)}`,
+      { bucket: bucketName }
+    );
+    return new Bucket(scope, `existing_${bucketLookup.id}`, {
+      _existingBucket: bucketLookup,
+    });
+  }
+
+  private readonly bucket: S3Bucket | DataAwsS3Bucket;
+  private readonly public: boolean = false;
   private readonly notificationTopics: S3BucketNotificationTopic[] = [];
   private readonly notificationDependencies: ITerraformDependable[] = [];
 
-  constructor(scope: Construct, id: string, props: cloud.BucketProps = {}) {
+  constructor(scope: Construct, id: string, props: BucketProps = {}) {
     super(scope, id, props);
-
+    if (props._existingBucket) {
+      this.bucket = props._existingBucket;
+      return;
+    }
```

Comment:
Suppose we wanted to expose whether a bucket is public or not as a public property:

```js
let bucket = new cloud.Bucket(public: true);
assert(bucket.public == true);
```

I'm wondering how we would do that in the current implementation?

(food for thought) I think the way a lot of AWS CDK code does this is they define an interface, and create a new class that satisfies:

```ts
interface IBucket extends IResource {
  ...
}

class Bucket {
  public static fromBucketName(scope: Construct, bucketName: string): IBucket {
    class BucketReference extends Construct implements IBucket {
      ...
    }
    return new BucketReference(scope, ...);
  }
}
```

Reply:
```js
public class Bucket {
  public static fromBucketName(scope: Construct, bucketName: string): Bucket {
    return new BucketReference(scope, id, bucketName);
  }
}

class BucketReference extends cloud.Bucket {
  get public() {
    throw new Error("not supported for referenced bucket");
  }
}
```

☝️ Something like this